### PR TITLE
Remove header button on mobile and push these into new floating btn

### LINKED ISSUE #22978: [BO - Responsive] Replace header button by a floatting button on mobile
#### Is your feature request related to a problem?

Header buttons are taking too much places, we want's to replace these by a floating button

Capture d’écran 2021-01-25 à 15 35 12

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We wants to get rid of header button as they use too much space on mobile, so we wants to push them to the floating button
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22978.
| How to test?      | Go on Product Lists page for example, see if on desktop view you you see the buttons, and on mobile view you see the floating button. PS: Recommended Modules can't be inside the button for the moment, the module itself needs to be updated as the markup is pushed using JS (which is a bad practice)
| Possible impacts? | Modules using this bar without using the hook

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22984)

image

### CODE REVIEW

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl [hunk 1]
```diff
@@ -120,5 +121,66 @@
...
+  <div class="btn-floating">
+    <button class="btn btn-primary collapsed" data-toggle="collapse" data-target=".btn-floating-container" aria-expanded="false">
+      <i class="material-icons">add</i>
+    </button>
+    <div class="btn-floating-container collapse">
+      <div class="btn-floating-menu">
+        {hook h='displayDashboardToolbarTopMenu'}
+        {foreach from=$toolbar_btn item=btn key=k}
+          {if $k != 'back' && $k != 'modules-list'}
+            {* TODO: REFACTOR ALL THIS THINGS *}
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/template/page_header_toolbar.tpl]: What do you mean by refactor?

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl [hunk 2]
```diff
@@ -120,5 +121,66 @@
...
+              data-placement="bottom"{/if}
+            >
+              {$btn.desc|escape}
+              {if !empty($btn.icon)}<i class="material-icons">{$btn.icon}</i>{/if}
+            </a>
+          {/if}
+        {/foreach}
+
+        {if isset($toolbar_btn['modules-list'])}
+          {* TODO: REFACTOR ALL THIS THINGS *}
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/template/page_header_toolbar.tpl]: What do you mean by refactor?

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl (hunk 1 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/template/page_header_toolbar.tpl]: I don't know, that's a copy paste from the top of the file, as it's the exact same markup, but on mobile it's inside the floating button

This is not me who wrote this line :/

**Review by atomiix — DISMISSED** [review]

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl (hunk 1 above)
Comment (NeOMakinG) [reply admin-dev/themes/new-theme/template/page_header_toolbar.tpl]: @PierreRambaud

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/template/page_header_toolbar.tpl]: Remove these lines if it's not needed and you don't know why they are here :sweat_smile:

Comment (PierreRambaud) [inline admin-dev/themes/new-theme/template/page_header_toolbar.tpl]:
```suggestion
```

File: admin-dev/themes/new-theme/template/page_header_toolbar.tpl (hunk 2 above)
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/template/page_header_toolbar.tpl]:
```suggestion
```

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by prestascott** [comment]
Nice! We just have to make sure the floating button options have the same design than bulk actions options so we keep the consistency in our back office. 

Like this:
image

**Comment by NeOMakinG** [comment]
> Nice! We just have to make sure the floating button options have the same design than bulk actions options so we keep the consistency in our back office.
> 
> Like this:
> image

Will be done inside next uikit release then, impossible for the moment, but this doesn't block this PR

**Comment by SD1982** [comment]
LGTM !!
Tested with browserstack on different emulated device and browser
**Samsung s20 Chrome**
https://drive.google.com/file/d/1K6lLTrouxtelxq32ihhOd4PcwFod3wLS/view?usp=sharing
**Samsung galaxy tab Firefox**
https://drive.google.com/file/d/1yWSVcx8lpbkfzLHPUpAGPBf8xqJ0HJL-/view?usp=sharing
**The break point on browser tools**
https://drive.google.com/file/d/1n2zZ_vXlKpaaGGCA-i0ACSO2cUPEF7aP/view?usp=sharing

As we saw together, i opened a new issue here #23016 for the aspect of the helpbox on the tablet format
Thanks @NeOMakinG !!

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @SD1982

**Pull request merged**